cql3/values.hh

```diff
@@ -240,10 +240,28 @@ public:
         return is_value();
     }
     bytes to_bytes() && {
-        switch (_data.index()) {
-        case 0:  return std::move(std::get<bytes>(_data));
-        default: return ::to_bytes(std::get<managed_bytes>(_data));
-        }
+        return std::visit(overloaded_functor{
+            [](bytes&& bytes_val) { return std::move(bytes_val); },
+            [](managed_bytes&& managed_bytes_val) { return ::to_bytes(managed_bytes_val); },
+            [](null_value&&) -> bytes {
+                throw std::runtime_error("to_bytes() called on raw value that is null");
+            },
+            [](unset_value&&) -> bytes {
+                throw std::runtime_error("to_bytes() called on raw value that is unset_value");
+            }
+        }, std::move(_data));
+    }
+    bytes_opt to_bytes_opt() && {
```

Comment:
It's better to return managed_bytes_opt, no? In case it's a large value.

Reply:
I assume you refer to the type of variable of `initcond` (the comment refers to the implementation of `to_bytes_opt()`, which probably should return managed_bytes, when `to_managed_bytes_opt()` exists).
It makes sense to change it to managed_bytes, but it would require a larger change and this patch only fixes the current implementation with just `bytes`